src/vs/workbench/contrib/terminalContrib/chatAgentTools/browser/executeStrategy/basicExecuteStrategy.ts

```diff
@@ -84,10 +89,11 @@ export class BasicExecuteStrategy implements ITerminalExecuteStrategy {
 			// Record where the command started. If the marker gets disposed, re-created it where
 			// the cursor is. This can happen in prompts where they clear the line and rerender it
 			// like powerlevel10k's transient prompt
-			let startMarker = store.add(xterm.raw.registerMarker());
-			store.add(startMarker.onDispose(() => {
+			this._startMarker = store.add(xterm.raw.registerMarker());
+			this._onDidCreateStartMarker.fire(this._startMarker);
+			store.add(this._startMarker.onDispose(() => {
 				this._log(`Start marker was disposed, recreating`);
-				startMarker = xterm.raw.registerMarker();
+				this._startMarker = xterm.raw.registerMarker();
```

Comment:
Considering it can change, let's make the even `onDidChangeStartMarker` and refire here